smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -302,17 +246,25 @@ def _on_iteration(self) -> None:
                     "Could not find model key or model.",
                 )
                 return
-            if request.model_key in self._cached_models:
+
+            # if request.model_key.descriptor not in self._feature_stores:
+            #     self._fs_factory(request.model_key.descriptor)
+            # todo: decide if we should load here or in _check_feature_stores.
+            # todo: should i raise error here?
```

Comment:
Looks like you put it in `_check_feature_stores`, which I like! I think as far as raising errors, we might want to use the `exception_handler` to report to the user that the validation step failed.

```
if not self._validate_request(request):
       exception_handler(
                    ValueError("Error validating the request"),
                    request.callback,
                    "Error validating the request.",
                )
       return
```

Reply:
Great catch. Done!